### PR TITLE
Add Inventec d6254qs d6556 platform suport and status led, sensors supports for all Inventec platform

### PR DESCRIPTION
**Pull request by klhaung**
[device][platform] d6254qs d6556 platform
[platform] status led, sensors on d6254qs 6556 d7032q28b d7054q28b d7264q28b

**- What I did**
* Add Inventec D6254QS,D6556 platform support
* Add status led, sensors support for all inventec platform (D6254QS, D6556, D7032Q28B, D7054Q28B, D7264Q28B)
* Bug fix in transceiver_monitor.py

**- How I did it**
* Follow Porting Guide to port D6254QS,D6556
* Follow Porting Guide to configure sensors.conf for all inventec platform (D6254QS, D6556, D7032Q28B, D7054Q28B, D7264Q28B)
* Add drivers and python scripts for monitoring system status and doing the corresponding actions for all inventec platform ((D6254QS, D6556, D7032Q28B, D7054Q28B, D7264Q28B))
* Add one more transceiver in transceiver_monitor.py

**- How to verify it**
* Build image, install on device platform to make sure it is worked properly.

**- Description for the changelog**

* Add Inventec D6254QS,D6556 platform support
* Add status led, sensors supports for all Inventec platform (D6254QS, D6556, D7032Q28B, D7054Q28B, D7264Q28B)

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by klhaung** [comment]
[device][platform] d6254qs d6556 d7264 platform
[platform] status led, sensors on d6254qs 6556 d7032q28b d7054q28b d7264q28b

- What I did
* Add Inventec D6254QS,D6556,D7264Q platform support
* Add status led, sensors support for all inventec platform (D6254QS, D6556, D7032Q28B, D7054Q28B, D7264Q28B)
* Bug fix in transceiver_monitor.py 

- How I did it
* Follow Porting Guide to port D6254QS,D6556,D7264Q
* Follow Porting Guide to configure sensors.conf for all inventec platform
* Add drivers and python scripts for monitoring system status and doing the corresponding actions 
* Add one more transceiver in transceiver_monitor.py

- How to verify it
* Build image, install on device platform to make sure it is worked properly.

- Description for the changelog
* Add Inventec  D6254QS,D6556,D7264Q platform support
* Add status led, sensors supports for all Inventec platform (D6254QS, D6556, D7032Q28B, D7054Q28B, D7264Q28B)

- A picture of a cute animal (not mandatory but encouraged)

**Comment by lguohan** [comment]
> port_flex_enable

not all attributes in 
this list.

[http://broadcom-switch.github.io/OpenNSL/doc/html/OPENNSL_CUSTOMIZING_OPENNSL.html](http://broadcom-switch.github.io/OpenNSL/doc/html/OPENNSL_CUSTOMIZING_OPENNSL.html)


---
Refers to: src/sonic-device-data/tests/permitted_list:142 in c131457. [](commit_id = c1314575891f70eed97900cc161d8144a142e8e9, deletion_comment = False)

**Comment by lguohan** [comment]
build break for broadcom platform.

**Comment by lguohan** [comment]
what are bcm56870_a0_cmh.pkg those files, zero bytes? Please remove.

File: device/inventec/x86_64-inventec_d6556-r0/INVENTEC-D6556/td3-d6556-48x25G-8x100G.config.bcm [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+portmap_127=125:100
+
+l2_mem_entries=32768
+l3_mem_entries=16384
+fpem_mem_entries=16384
+l2xmsg_mode=1
+
+pdma_dv_free_count=64
+pdma_dv_free_size=320
+pdma_descriptor_prefetch_enable=1
```
Comment (lguohan) [inline device/inventec/x86_64-inventec_d6556-r0/INVENTEC-D6556/td3-d6556-48x25G-8x100G.config.bcm]: why do you need these pdma settings?

**Comment by lguohan** [comment]
>  *  lpc_ich.c - LPC interface for Intel ICH

4.9 kernel has this file, please reuse the module in the kernel. if you need to modify, please patch the kernel.

---
Refers to: platform/broadcom/sonic-platform-modules-inventec/d6556/modules/lpc_ich.c:2 in c131457. [](commit_id = c1314575891f70eed97900cc161d8144a142e8e9, deletion_comment = False)

**Comment by lguohan** [comment]
please add license for every file.

**Comment by klhaung** [comment]
#1 For build break for broadcom platform:
From console log, look like it broke becasue of kernel change from 3.16 to 4.9. We will check how to make it compiled on kernel 4.9.

#2 For lpc_ich.c
We did all these fixes before 7/E and kernel 3.6 doesn't have this module by default. We will study how to port new platform again on kernel 4.9

#3 What does this kernel thread do
use to check psu modules and fan modules and send uevent to notify user-space application for doing corresponding actions. It is better to do them in driver in case the user-space application is stopped by some unknown reason. 

#4 why do you need these pdma settings?

We follows Broadcom's suggestion to configure TD3. The example bcm files are config_td3_128x25G_pwr.bcm and config_td3_tr514_os.bcm, not quite understand what they are used for from application point of view.

# not all attributes in this list.

Good point. Look like Broadcom doesn't announce all the attributes for TD3 yet. Without these three attributes, cancun_dir, led_fw_dir, ifp_inports_support_enable, we cannot bring up SONiC on TD3 successfully. Without the other three attributes, we got warning messages when syncd bring up.

**Comment by klhaung** [comment]
Close this PR becaus the fixes we did was done and verified on kernel 3.6.

**Pull request closed without merging**